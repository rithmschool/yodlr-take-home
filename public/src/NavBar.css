nav {
  display: flex;
  background-color: purple;
  padding-left: 0;
  padding-top: 1%;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
}

.nav-item {
  display: inline-block;
  width: 5rem;
  text-align: center;
}

nav a {
  display: inline-block;
  color: #f0f2f5;
  text-decoration: none;
}

nav a.active {
  display: inline-block;
  color: #f0f2f5;
  font-weight: bold;
  border-bottom: 5px;
  box-sizing: border-box;
}

nav a.active:hover {
  color: #f0f2f5 !important;
}

nav a:hover {
  display: inline-block;
  text-decoration: none;
  color: #f0f2f5;
}

nav a.active.navbar-brand {
  float: left;
  display: inline;
  border-bottom: none;
}

nav a:not(.navbar-brand):hover {
  font-weight: bold;
}

.login{
  display: inline-block;
  white-space: nowrap;
  writing-mode: horizontal-tb;
}

nav .navbar-brand {
  float: left;
  display: inline;
  vertical-align: middle;
  font-size: x-large;

}

.ml-auto, .mx-auto {
  margin-left: auto!important;
}

ul {
  margin-block-start: 1em;
    margin-block-end: 1em;
    margin-inline-start: 2em;
    margin-inline-end: 2.5em;
    padding-inline-start: 40px;
}