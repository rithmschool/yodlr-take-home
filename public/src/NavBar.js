import React, { useContext } from 'react';
import {
  Navbar,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';
import {CurrentUserContext} from './YodlrContext';
import './NavBar.css';

function NavBar ({ logout }) {
  const currentUser = useContext(CurrentUserContext);

  // display navbar at top of page, provides access to admin page if logged in
  return (
    <div>
      <Navbar color="light" light expand="md">
        <NavbarBrand href="/">Yodlr</NavbarBrand>
          <Nav className="mr-auto" navbar>
            <NavItem>
              <NavLink href="/signup">Register</NavLink>
            </NavItem>
            {(currentUser) ? 
            <NavItem>
              <NavLink href="/admin">Admin</NavLink>
            </NavItem> : 
            <NavItem>
            <NavLink href="/login">Login</NavLink>
          </NavItem>}
          {(currentUser) ? <NavItem>
            <NavLink to="/" onClick={logout}>Logout</NavLink>
          </NavItem> : <div/>}
          </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;