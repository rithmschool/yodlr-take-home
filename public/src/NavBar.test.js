import React from "react";
import { MemoryRouter } from 'react-router-dom';
import { render } from "@testing-library/react";
import NavBar from './NavBar';

it("renders without crashing", function() {
  render(<MemoryRouter><NavBar /></MemoryRouter>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<MemoryRouter><NavBar /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});