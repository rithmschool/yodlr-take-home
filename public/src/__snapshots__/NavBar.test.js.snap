// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Yodlr
      </a>
      <ul
        class="mr-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/signup"
          >
            Register
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
        <div />
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
